### PR TITLE
Add page about formatting code

### PR DESCRIPTION
**Pull request by kwalrath**
Gets rid of our last sidenav TBD!

### CODE REVIEW

File: src/_guides/language/formatting.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+title: Formatting code
+description: Use dartfmt to format your code, and follow Effective Dart guidelines for what dartfmt doesn't cover.
+---
+
+As [Effective Dart][] says, when it comes to things like formatting,
+arguments about which is better are subjective and impossible to resolve.
```
Comment (mit-mit) [inline src/_guides/language/formatting.md]: maybe s/impossible/time consuming/

File: src/_guides/language/formatting.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
+title: Formatting code
+description: Use dartfmt to format your code, and follow Effective Dart guidelines for what dartfmt doesn't cover.
```
Comment (mit-mit) [inline src/_guides/language/formatting.md]: nit: Lots of different line lenghts. Maybe auto-wrap everything at 80 cols? (especially given this is a page about automating formatting ;-)

File: src/_guides/language/formatting.md (hunk 1 above)
Comment (kwalrath) [reply src/_guides/language/formatting.md]: I'm just quoting Effective Dart here. :) But yeah, agreed.

File: src/_guides/language/formatting.md (hunk 2 above)
Comment (kwalrath) [reply src/_guides/language/formatting.md]: You sound like Bob! I prefer breaking on phrases, to minimize diffs caused by whitespace. It's like breaking after a semicolon... or whatever. :)

**Pull request merged**